build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -46,12 +46,14 @@ public class CompileTargets
             { "osx.10.11-x64", "osx.10.10-x64" },
             { "ubuntu.14.04-x64", "ubuntu.14.04-x64" },
             { "ubuntu.16.04-x64", "ubuntu.16.04-x64" },
+            { "ubuntu.16.10-x64", "ubuntu.16.10-x64" },
             { "centos.7-x64", "rhel.7-x64" },
             { "rhel.7-x64", "rhel.7-x64" },
             { "rhel.7.2-x64", "rhel.7-x64" },
             { "debian.8-x64", "debian.8-x64" },
             { "fedora.23-x64", "fedora.23-x64" },
-            { "opensuse.13.2-x64", "opensuse.13.2-x64" }
+            { "opensuse.13.2-x64", "opensuse.13.2-x64" },
+            { "opensuse.42.1-x64", "opensuse.42.1-x64" }
```

Comment:
We need docker files for these new OSes. Otherwise how are they going to be built in CI? Also, you need to update groovy.netci to add these OSes to Jenkins.